scripts/pre-commit.sh

```diff
@@ -20,31 +20,83 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
+source "$(dirname "${BASH_SOURCE}")/util.sh"
+
+function build-binaries() {
+  go build -o bin/controller-manager ./cmd/controller-manager
+  go build -o bin/kubefed2 ./cmd/kubefed2
+}
+
+function run-integration-tests() {
+  # Ensure the test binaries are in the path.
+  export TEST_ASSET_PATH="${base_dir}/bin"
+  export TEST_ASSET_ETCD="${TEST_ASSET_PATH}/etcd"
+  export TEST_ASSET_KUBE_APISERVER="${TEST_ASSET_PATH}/kube-apiserver"
+  go test -v ./test/integration
+  rc=$((rc || $?))
+  return ${rc}
+}
+
+function launch-minikube-cluster() {
+  # Move crictl to system path as expected by kubeadm.
+  sudo mv ${TEST_ASSET_PATH}/crictl /usr/bin/
+  sudo bash -c "export PATH=${PATH}; minikube start -p cluster1 --kubernetes-version v1.11.0 --vm-driver=none --v=4"
+
+  # Change ownership of .kube and .minikube config to use without sudo.
+  sudo chown -R $USER $HOME/.kube
+  sudo chgrp -R $USER $HOME/.kube
+
+  sudo chown -R $USER $HOME/.minikube
+  sudo chgrp -R $USER $HOME/.minikube
+}
+
+# Wait for the kubernetes API server to be ready.
+function kube-apiserver-ready() {
+  local result="$(kubectl -n kube-system get pod kube-apiserver-minikube -o jsonpath='{.status.conditions[?(@.type == "Ready")].status}' 2> /dev/null)"
```

Comment:
Calling the apiserver to get the pod status of the apiserver seems strange, am I missing something?  Regardless, consider calling `kubectl get --raw=/healthz` instead.

Reply:
After updating to use this, the cluster readiness comes back as `ok` almost immediately. This is somewhat concerning because several `kube-system` components are still not ready. I'd have to investigate it further, but for now it seems to be working.

Reply:
Would using this additionally be of any value?
`kubectl get po -n kube-system  --field-selector="status.phase!=Running"`
which should return
`No resources found.`

I wouldn't fret too much on it, if its working nevertheless; we would anyways start to see test failures if the cluster isn't healthy.

Reply:
I think this would work if all pods were launched. But there will be pods that aren't launched yet so their status may not even exist.